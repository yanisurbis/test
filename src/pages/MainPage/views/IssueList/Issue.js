import React, { PropTypes as t } from 'react'
import { Link } from 'react-router-dom'
import { Item } from 'semantic-ui-react'
import timeago from 'timeago.js'

// import './Issue.css'

function Issue({ issue }) {
  const { number, title, created_at, fullRepoName, user } = issue
  const { avatar_url, html_url, login } = user

  return (
    <Item>
      <Item.Image size='mini' src={avatar_url} />

      <Item.Content>
        <Item.Header>
          <Link
            to={{
              pathname: `${fullRepoName}/issues/${number}`,
              // we pass index of issue through props to retrieve data from redux store later
              // we use here react router v5 feature here
              // state: { index },
            }}
          >
            {title}.
          </Link>
        </Item.Header>
        <Item.Meta>
          #{number} - {new timeago().format(created_at)} by <a href={html_url} target="_blank">{login}</a>
        </Item.Meta>
      </Item.Content>
    </Item>
  );
}

Issue.propTypes = {
  issue: t.shape({
    title: t.string.isRequired,
    number: t.number.isRequired,
    created_at: t.string.isRequired,
    user: t.shape({
      avatar_url: t.string.isRequired,
      login: t.string.isRequired,
      html_url: t.string.isRequired,
    })
  })
}

export default Issue
