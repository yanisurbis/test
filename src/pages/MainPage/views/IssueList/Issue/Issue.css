.issue
{
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid #ffc300;
  margin-top: 20px;
  display: flex;
  border-top: 4px solid #ffc300;
  margin-top: 18px;
}

.img
{
  width: 72px;
  height: 72px;
}
