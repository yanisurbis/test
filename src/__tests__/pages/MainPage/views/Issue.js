/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Issue from 'code/pages/MainPage/views/IssueList/Issue'

const createProps = () => ({
  issue: {
    title: 'sample',
    number: 1,
    created_at: (new Date()).toString(),
    user: {
      avatar_url: 'sample',
      login: 'sample',
      html_url: 'sample',
    }
  }
});

test('<Issue /> renders correctly', () => {
  const props = createProps();
  const wrapper = shallow(
    <Issue {...props} />
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
