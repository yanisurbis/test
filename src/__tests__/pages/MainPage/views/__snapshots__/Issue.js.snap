exports[`test <Issue /> renders correctly 1`] = `
<Item>
  <ItemImage
    size="mini"
    src="sample" />
  <ItemContent>
    <ItemHeader>
      <Link
        replace={false}
        to={
          Object {
            "pathname": "undefined/issues/1",
          }
        }>
        sample
        .
      </Link>
    </ItemHeader>
    <ItemMeta>
      #
      1
       - 
      just now
       by 
      <a
        href="sample"
        target="_blank">
        sample
      </a>
    </ItemMeta>
  </ItemContent>
</Item>
`;
